unit-tests/func/reflectivity/l500/test-auto-set-options.cpp

```diff
@@ -0,0 +1,192 @@
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+
+    ds.stop();
+    ds.close();
+}
+
+TEST_CASE("Automatically turn on MUR option - stays on", "[reflectivity]")
+{
+    auto ds
+        = find_first_supported_depth_sensor_or_exit("L515", RS2_OPTION_ENABLE_IR_REFLECTIVITY);
+
+    // Verify alt IR is off
+    if (ds.supports(RS2_OPTION_ALTERNATE_IR))
+        ds.set_option(RS2_OPTION_ALTERNATE_IR, 0);
+
+    ds.set_option(RS2_OPTION_SENSOR_MODE, RS2_SENSOR_MODE_VGA);
+    ds.set_option(RS2_OPTION_VISUAL_PRESET, RS2_L500_VISUAL_PRESET_MAX_RANGE);
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE, 0));
+    // Verify automatically turn on of MUR option
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+
+    // Verify automatically turn off of MUR option
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 0));
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+
+    // Turn on MUR
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE, 1));
+
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 0));
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    // Verify MUR stays on
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+}
+
+
+TEST_CASE("Automatically turn on/off MUR option", "[reflectivity]")
+{
+    auto ds
+        = find_first_supported_depth_sensor_or_exit("L515", RS2_OPTION_ENABLE_IR_REFLECTIVITY);
+
+    // Verify alt IR is off
+    if (ds.supports(RS2_OPTION_ALTERNATE_IR))
+        ds.set_option(RS2_OPTION_ALTERNATE_IR, 0);
+
+    ds.set_option(RS2_OPTION_SENSOR_MODE, RS2_SENSOR_MODE_VGA);
+    ds.set_option(RS2_OPTION_VISUAL_PRESET, RS2_L500_VISUAL_PRESET_MAX_RANGE);
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE, 0));
+
+    auto depth_vga_profile
+        = get_profile_by_stream_parameters(ds, RS2_STREAM_DEPTH, RS2_FORMAT_Z16, 640, 480, 30, 0);
+
+    REQUIRE(depth_vga_profile);
+
+    ds.open(depth_vga_profile);
+    ds.start([](rs2::frame f) {});
+
+    // Verify automatically turn on of MUR option
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 1.f));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    CHECK(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+
+    // Verify automatically turn off of MUR option
+    CHECK_NOTHROW(ds.set_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY, 0));
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY));
+    CHECK_FALSE(ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE));
+
```

Comment:
What about the other checks in the prev test? MUR=1, Refl->1, MUR == 1, Refl->0, MUR == 1

Reply:
👍